specs/DefaultDownloadDialog.md

```diff
@@ -0,0 +1,249 @@
+Default Download Dialog Layout
+===
+
+# Background
+This API gives you the ability to change the position of the default download dialog
+relative to the webview window, programmatically open and close the dialog, and
+make changes in response to the dialog opening or closing.
+
+You can anchor the default download dialog to a button in your application by setting its
+position below your button, and alternating open and close calls when the button is
+clicked on.
+
+For more complex customizations, you can use the Download APIs -
+[DownloadStarting](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2downloadstartingeventargs?view=webview2-1.0.961.33)
+ and [DownloadOperation](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2downloadoperation?view=webview2-1.0.961.33) - to build custom download UI.
+
+# Examples
+```cpp
+// Subscribe to the `IsDefaultDownloadDialogOpenChanged` event
+// to make changes in response to the default download dialog
+// opening or closing. For example, if the dialog is anchored
+// to a button in the application, the button can be updated to
+// appear selected when the dialog is opened.
+CHECK_FAILURE(m_webView2_6->add_IsDefaultDownloadDialogOpenChanged(
+    Callback<ICoreWebView2IsDefaultDownloadDialogOpenChangedEventHandler>(
+        [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT {
+        BOOL isOpen;
+        m_webView2_6->get_IsDefaultDownloadDialogOpen(&isOpen);
+        if (isOpen)
+        {
+            // Here the developer can make updates in response to the
+            // default download dialog opening.
+        }
+        else
+        {
+            // Here the developer can make updates in response to the
+            // default download dialog closing.
+        }
+        return S_OK;
+        })
+        .Get(),
+    &m_isDefaultDownloadDialogOpenChangedToken));
+
+void ViewComponent::ToggleDefaultDownloadDialog()
+{
+    if (m_webView2_6)
+    {
+        BOOL isOpen;
+        m_webView2_6->get_IsDefaultDownloadDialogOpen(&isOpen);
+        if (isOpen)
+        {
+            m_webView2_6->CloseDefaultDownloadDialog();
+        }
+        else
+        {
+            m_webView2_6->OpenDefaultDownloadDialog();
```

Comment:
Should we model this as a read/write property?

Reply:
Discussion: despite XAML precedence feels more natural as methods than property. Leave as is.

Reply:
Please explicitly note in documentation that calling Open/Close programmatically will explicitly raise the property changed event (iff the property actually changes value).